src/parser/jobs/blu/modules/ColdFog.tsx

```diff
@@ -0,0 +1,146 @@
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+// Tiered suggestion severities
+const COLD_SEVERITY = {
+	DROPPED_COLD_FOG: {
+		1: SEVERITY.MAJOR,
+	},
+	DROPPED_WHITE_DEATH_CASTS: {
+		1: SEVERITY.MEDIUM,
+	},
+	INEFFECTIVE_COLD_FOG: {
+		1: SEVERITY.MAJOR,
+	},
+}
+
+const COLD_FOG_MINIMUM_VIABLE_CASTS = 2
+const COLD_COLD_IDEAL_CASTS = 6
+
+interface ColdFogWindow {
+	whiteDeathCasts: number
+}
+
+export class ColdFog extends Analyser {
+	static override handle = 'coldfog'
+	static override title = t('blu.cold_fog.title')`Cold Fog`
+
+	@dependency private actors!: Actors
+	@dependency private data!: Data
+	@dependency private suggestions!: Suggestions
+
+	private touchOfFrostProcs = 0
+	private coldFogCasts = 0
+	private currentHook?: EventHook<Events['action']>
+	private touchOfFrostHistory = new History<ColdFogWindow>(
+		() => ({
+			whiteDeathCasts: 0,
+		})
+	)
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.TOUCH_OF_FROST.id), this.onApplyTouchOfFrost)
+		this.addEventHook(playerFilter.type('statusRemove').status(this.data.statuses.TOUCH_OF_FROST.id), this.onRemoveTouchOfFrost)
+
+		this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.COLD_FOG.id), this.onApplyColdFog)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onApplyTouchOfFrost(event: Events['statusApply']) {
+		this.touchOfFrostHistory.openNew(event.timestamp)
+		if (this.currentHook !== undefined) { return }
+
+		// Start tracking to see if they got the six White Death GCDs during the window
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.currentHook = this.addEventHook(playerFilter.action(this.data.actions.WHITE_DEATH.id).type('action'), this.onWhiteDeath)
+	}
+	private onRemoveTouchOfFrost() {
+		this.touchOfFrostHistory.closeCurrent(this.parser.pull.timestamp + this.parser.pull.duration)
+		if (this.currentHook !== undefined) {
+			this.removeEventHook(this.currentHook)
+		}
+	}
+
+	private onWhiteDeath(event: Events['action']) {
+		const current = this.touchOfFrostHistory.getCurrent()
+		if (current == null) { return }
+
+		if (event.action === this.data.actions.WHITE_DEATH.id) {
+			current.data.whiteDeathCasts++
+		}
+	}
+
+	private onApplyColdFog() {
+		this.coldFogCasts++
+	}
+
+	private onComplete() {
+
+		// Wiffled Cold Fog
+		const droppedColdFogs = this.coldFogCasts - this.touchOfFrostHistory.entries.length
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.data.actions.COLD_FOG.icon,
+			content: <Trans id="blu.coldfog.dropped.content">
+				Any damage taken while the <DataLink status="COLD_FOG"/> effect is active grants the <DataLink status="TOUCH_OF_FROST"/>
+				buff, allowing the use of <DataLink action="WHITE_DEATH" /> for 15 seconds.
```

Comment:
If you want the newlines to actually show in the end result output, don't forget to include the </br> elements at the end of each line. As it is, the next line is just going to directly append itself to the previous and you'll lose some of the intended whitespace.

Reply:
Yup, had a missing `br`, patched in f1fc88138da52e821897195b74827ce058a041ce